### PR TITLE
Persistence Component: Introduce support for DuckDB 

### PR DESCRIPTION
**Pull request by kumuwu**
#### What type of PR is this?

- Improvement

#### What does this PR do / why is it needed ?

Add support for DuckDB, supported ingest modes include:
- Append Only
- Unitemporal Snapshot
- Unitemporal Delta

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalGeneratorAbstract.java [hunk 1]
```diff
@@ -225,6 +225,11 @@ GeneratorResult generateOperations(Datasets datasets, Resources resources)
 
     GeneratorResult generateOperations(Datasets datasets, Resources resources, Planner planner, IngestMode ingestMode)
     {
+        if (!relationalSink().isIngestModeSupported(ingestMode))
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalGeneratorAbstract.java:228]: Should we do it even earlier -- may be like a validation using @Check  ?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/jdbc/DuckDBJdbcHelper.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Override
+    public Dataset constructDatasetFromDatabase(Dataset dataset, TypeMapping typeMapping, boolean escape)
+    {
+        String tableName = dataset.datasetReference().name().orElseThrow(IllegalStateException::new);
+        String schemaName = dataset.datasetReference().group().orElse(null);
+        String databaseName = dataset.datasetReference().database().orElse(null);
+
+        String escapedTableName = tableName.replace("_", "\\_");
```
Comment (prasar-ashutosh) [inline legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/jdbc/DuckDBJdbcHelper.java:77]: Is this needed in Duck DB ?

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalGeneratorAbstract.java (hunk 1 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/src/main/java/org/finos/legend/engine/persistence/components/relational/api/RelationalGeneratorAbstract.java]: Sure great idea, changed!

File: legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/jdbc/DuckDBJdbcHelper.java (hunk 2 above)
Comment (kumuwu) [reply legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-duckdb/src/main/java/org/finos/legend/engine/persistence/components/relational/duckdb/jdbc/DuckDBJdbcHelper.java]: Have removed this part from this class for now - will properly clean up the whole logic for escaping for snowflake in a separate PR

**Review by epsstan — APPROVED** [review]

**Pull request merged**